cassandra/io/asyncorereactor.py

```diff
@@ -279,6 +279,9 @@ def _cleanup(self):
         if not self._thread:
             return
 
+        # The loop shouldn't be woken up from here onwards since it won't be running
+        self._loop_dispatcher._notified = True
```

Comment:
I don't love this, but I'd rather do it this way than making more complex the logic `_AsyncoreUDPDispatcher` and `_AsyncorePipeDispatcher` by adding for example a `closed` variable.